app/utils/service_client/middleware/body_encoder.rb

```diff
@@ -97,7 +98,12 @@ def leave(ctx)
         encoder = encoder_by_content_type(headers["Content-Type"]) || @_request_encoder
 
         begin
-          ctx[:res][:body] = encoder[:encoder].decode(body)
+          ctx[:res][:body] =
+            if @_decode_response
+              encoder[:encoder].decode(body)
+            else
+              body
+            end
```

Comment:
This functionality could be implemented by setting the encoder's decode method to identity function at configuration time in case the @_decode_response is false. But it comes down to stylistic choice, one more if per request is unlikely to make a difference.